src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -479,3 +426,90 @@ export const AddressVerificationFlowQueryRenderer: React.FC<{
     />
   )
 }
+
+const fallbackFromFormValues = (address: AddressValues): AddressOption => {
+  return {
+    key: USER_INPUT,
+    recommended: false,
+    lines: [
+      address.addressLine1,
+      address.addressLine2 || null,
+      `${address.city}, ${address.region} ${address.postalCode}`,
+      address.country,
+    ].filter(Boolean) as string[],
+
+    address: {
+      addressLine1: address.addressLine1,
+      addressLine2: address.addressLine2,
+      region: address.region,
+      country: address.country,
+      city: address.city,
+      postalCode: address.postalCode,
+    },
+  }
+}
+
+const useAddressVerificationTracking = () => {
+  const { trackEvent } = useTracking()
+  const { contextPageOwnerId } = useAnalyticsContext()
+  const { user } = useSystemContext()
+  const userId = user?.id
+
+  return {
+    trackViewedModal: useCallback(
+      ({ subject }: { subject: string }) => {
+        trackEvent({
+          action: ActionType.validationAddressViewed,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: OwnerType.ordersShipping,
+          context_page_owner_id: contextPageOwnerId,
+          user_id: userId,
+          flow: "user adding shipping address",
+          subject,
+        } as ValidationAddressViewed)
+      },
+      [contextPageOwnerId, trackEvent, userId]
+    ),
+
+    trackSelectedAddress: useCallback(
+      ({
+        label,
+        option,
+        subject,
+      }: {
+        label: string
+        option: string
+        subject: string
+      }) => {
+        const event: ClickedValidationAddressOptions = {
+          action: ActionType.clickedValidationAddressOptions,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: OwnerType.ordersShipping,
+          context_page_owner_id: contextPageOwnerId!,
+          user_id: userId!,
+          subject,
+          option,
+          label,
+        }
+        trackEvent(event)
+      },
+      [contextPageOwnerId, trackEvent, userId]
+    ),
+
+    trackClosedModal: useCallback(
+      ({ label, subject }: { label: string; subject: string }) => {
+        const event: ClickedCloseValidationAddressModal = {
+          action: ActionType.clickedCloseValidationAddressModal,
+          context_module: ContextModule.ordersShipping,
+          context_page_owner_type: OwnerType.ordersShipping,
+          context_page_owner_id: contextPageOwnerId!,
+          option: "close",
```

Comment:
"close" is not a valid option from the [spreadsheet](https://docs.google.com/spreadsheets/d/1c65vugkF2C4YuXwBXno8aqsBh4B2fTwxiLmgY_MhjWc/edit#gid=0). Looks like this needs to be dynamic.

![Screen Shot 2023-08-25 at 1 46 12 PM](https://github.com/artsy/force/assets/796573/97b9af21-042c-495a-8daf-05446efd3fe5)

Reply:
Correct, fixed with a type coercion for now, will ticket updating cohesion to permit this.